### ISSUE TITLE
[ER] catch some wrong copy_from_slice statically

### ISSUE THREAD

**Issue by leonardo-m** [rust-lang/rust#76476]
I think we should catch some `slice::copy_from_slice` length mismatch in basic cases like:

```rust
fn main() {
    let mut a = [0_u32; 10];
    let b = [0_u32; 20];
    a.copy_from_slice(&b);
}
```

**Comment by leonardo-m** [comment]
In alternative this could go in Clippy.
